var/spack/repos/builtin/packages/talloc/package.py

```diff
@@ -13,4 +13,8 @@ class Talloc(AutotoolsPackage):
     homepage = "https://talloc.samba.org"
     url      = "https://www.samba.org/ftp/talloc/talloc-2.1.9.tar.gz"
 
+    version('2.3.1', sha256='ef4822d2fdafd2be8e0cabc3ec3c806ae29b8268e932c5e9a4cd5585f37f9f77')
+    version('2.3.0', sha256='75d5bcb34482545a82ffb06da8f6c797f963a0da450d0830c669267b14992fc6')
     version('2.1.9', sha256='f0aad4cb88a3322207c82136ddc07bed48a37c2c21f82962d6c5ccb422711062')
+
+    extends('python@3:')
```

Comment:
Does `extends` handle versions correctly? If not you'll need to use extends and depends_on

Reply:
Sorry for replying so late due to my vacation. 

> Does extends handle versions correctly? If not you'll need to use extends and depends_on

Yes, It can handle version correctly.
If I use the code `extends('python@:2.9')`, it will depend on `python@2.7`, otherwise it will depend on `python@3.8`